lib/vocabularies/code.ts

```diff
@@ -90,8 +92,23 @@ export function callValidateCode(
   return context !== nil ? _`${func}.call(${context}, ${args})` : _`${func}(${args})`
 }
 
-export function usePattern({gen, it: {opts}}: KeywordCxt, pattern: string): Name {
+export function usePattern({gen, it: {self, opts}}: KeywordCxt, pattern: string): Name {
   const u = opts.unicodeRegExp ? "u" : ""
+  const useRe2 = opts.useRe2
+  if (u === "u" && useRe2) {
+    try {
+      const engine = new Re2(pattern)
+      return gen.scopeValue("pattern", {
+        key: pattern,
+        ref: engine,
+        code: _`new ${useFunc(gen, re2)}(${pattern})`,
+      })
+    } catch (e) {
+      self.logger.log(
+        "Warning: Fallback to default regex engine. Please verify the safeness of your regex"
```

Comment:
If this is left in, it could be clearer: "Warning: Your regex is not supported by RE2. Falling back to ..."

Reply:
Warning message changed to be more explanatory